algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/prim/Vertex.java

```diff
@@ -0,0 +1,103 @@
+package com.baeldung.algorithms.prim;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.commons.math3.util.Pair;
+
+public class Vertex {
+
+    private String label = null;
+    private Map<Vertex, Edge> edges = new HashMap<>();
+    private boolean isVisited = false;
+
+    public Vertex(String label) {
+        this.label = label;
+    }
+
+    public String getLabel() {
+        return label;
+    }
+
+    public void setLabel(String label) {
+        this.label = label;
+    }
+
+    public Map<Vertex, Edge> getEdges() {
+        return edges;
+    }
+
+    public void addEdge(Vertex vertex, Edge edge) {
+        this.edges.put(vertex, edge);
+    }
+
+    public boolean isVisited() {
+        return isVisited;
+    }
+
+    public void setVisited(boolean visited) {
+        isVisited = visited;
+    }
+
+    public Pair<Vertex, Edge> nextMinimum() {
+        Edge nextMinimum = new Edge(Integer.MAX_VALUE);
+        Vertex nextVertex = this;
+        Iterator it = edges.entrySet()
+            .iterator();
+        while (it.hasNext()) {
+            Map.Entry pair = (Map.Entry) it.next();
+            if (!((Vertex) pair.getKey()).isVisited()) {
+                if (!((Edge) pair.getValue()).isIncluded()) {
+                    if (((Edge) pair.getValue()).getWeight() < nextMinimum.getWeight()) {
+                        nextMinimum = (Edge) pair.getValue();
+                        nextVertex = (Vertex) pair.getKey();
+                    }
+                }
+            }
+        }
```

Comment:
A typed iterator here also could make the code more readable since we wouldn't have to cast the same thing three times during each iteration of the loop.

Reply:
fixed